### ISSUE TITLE
custom dataset

### ISSUE THREAD

**Issue by songWell** [qiaoyu-zheng/RP3D-Diag#6]
hello, you have done great work! I am a newcomer, and I have DCM format CT and MR data. How can I use predict to process my data? How can I convert it to the format used by your model. Is there any relevant guide? 

I look forward to your reply, thanks~

**Comment by qiaoyu-zheng** [comment]
For a series of DCM data, you can firstly convert it to the python array in the shape of [H, W, D]，then you can directly use the data normalization code in this project, which will change the shape into 256, 256, 32 and conduct normalization.

**Comment by WNjun** [comment]
> For a series of DCM data, you can firstly convert it to the python array in the shape of [H, W, D]，then you can directly use the data normalization code in this project, which will change the shape into 256, 256, 32 and conduct normalization.

@qiaoyu-zheng Hi, thank you for your kind guidance! I have the same request regarding DCM format MRI data, and I'm a bit lost in the code due to the numerous 'Dataset' subclasses in dataset.py. Could you please point me to the specific part of the code I should use for normalization? Any help would be greatly appreciated!

**Comment by qiaoyu-zheng** [comment]
@WNjun because my data is npy format, so my normalization code is used for array. But for DCM data, you can use python packages such as pydicom to convert it to the array in the shape of H W D, and then you can use the resample code:

def resample_image(image, tS, tD):
    depth = image.shape[0]
    height = image.shape[-2]
    width = image.shape[-1]
    # assert height == width
    size = height

    if size == tS and width == tS and depth <= tD:
        output_tensor = image
    else:
        output_tensor = torch.nn.functional.interpolate(image, size=(tS, tS), mode='bilinear', align_corners=False)
        if depth > tD:
            step = depth / tD
            indices = indices = torch.arange(0, depth, step).long()
            indices = indices[:tD]
            output_tensor = output_tensor[indices]
    return output_tensor

**Comment by WNjun** [comment]
@qiaoyu-zheng Thank you for your prompt reply! That was very helpful. One more small detail I'd like to confirm: when converting the DCM files to numpy arrays, do I need to preprocess them, such as removing outliers or applying Nyul normalization?
